internal/api/mfa.go

```diff
@@ -686,6 +681,38 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 	metering.RecordLogin(string(models.MFACodeLoginAction), user.ID)
 
 	return sendJSON(w, http.StatusOK, token)
+}
+
+func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	factor := getFactor(ctx)
+	config := a.config
+
+	params := &VerifyFactorParams{}
+	if err := retrieveRequestParams(r, params); err != nil {
+		return err
+	}
+
+	switch factor.FactorType {
+	case models.Phone:
+		if !config.MFA.Phone.VerifyEnabled {
+			return unprocessableEntityError(ErrorCodeMFAPhoneEnrollDisabled, "MFA verification is disabled for Phone")
+		}
+		if params.Code == "" {
+			return badRequestError(ErrorCodeValidationFailed, "Code needs to be non-empty")
+		}
```

Comment:
is there a reason why this validation needs to be handled separately in each case?

Reply:
Initial consideration was for additional factors which don't require a Code (e.g. WebAuthn)

but I guess we can move it since that seems to be  a while out